xgcm/grid_ufunc.py

```diff
@@ -1,9 +1,16 @@
 import re
 import string
+from typing import TYPE_CHECKING, Any, Callable, List, Mapping, Sequence, Tuple, Union
 
 import numpy as np
 import xarray as xr
 
+if TYPE_CHECKING:
+    # Avoids circular references when type checking
+
+    from .grid import Grid
```

Comment:
I'm a bit confused by this. What is the circular reference? Does `grid` import this module?

Could it possibly be avoided by [`from __future__ import annotations`](https://docs.python.org/3/whatsnew/3.7.html#pep-563-postponed-evaluation-of-annotations)?

Reply:
> Does grid import this module?

Yes - this refactor will mean that e.g. `grid.diff` calls `apply_as_grid_ufunc`, so `grid` has to import this module.

> Could it possibly be avoided by from __future__ import annotations?

That allows me to swap out `"Grid"` for `Grid`, but I still can't remove the `if TYPE_CHECKING` without causing a circular reference error...